modules/mockk/src/commonMain/kotlin/io/mockk/impl/restrict/RestrictMockkConfiguration.kt

```diff
@@ -14,10 +14,12 @@ class RestrictMockkConfiguration(propertiesLoader: PropertiesLoader = DefaultPro
         userDefinedRestrictedTypes = loadRestrictedTypesFromConfig(properties)
 
         restrictedTypes = DEFAULT_RESTRICTED_CLAZZ + userDefinedRestrictedTypes
-        throwExceptionOnBadMock = loadThrowExceptionSetting(properties)
+        throwExceptionOnBadMock = loadThrowExceptionSetting(properties) ||
+            loadThrowExceptionSystemProperty()
```

Comment:
I don't think `||` is the right combiner here, it should be `system ?: setting`.

Suppose the properties file contains:
`mockk.throwExceptionOnBadMock=true`
and you run
`gradlew -Pmockk.throwExceptionOnBadMock=false`
what would you expect?

I would run that command explicitly because "I just want to see the tests run without this check".